ui/components/app/toast-master/utils.ts

```diff
@@ -46,11 +48,22 @@ export function setShowNftDetectionEnablementToast(
 }
 
 export function setSwitchedNetworkNeverShowMessage() {
-  submitRequestToBackground('setSwitchedNetworkNeverShowMessage', [true]);
+  submitRequestToBackgroundAndCatch('setSwitchedNetworkNeverShowMessage', [
+    true,
+  ]);
 }
 
 export function setSurveyLinkLastClickedOrClosed(time: number) {
-  return async () => {
-    await submitRequestToBackground('setSurveyLinkLastClickedOrClosed', [time]);
-  };
+  submitRequestToBackgroundAndCatch('setSurveyLinkLastClickedOrClosed', [time]);
+}
+
+// May move this to a different file after discussion with team
+export function submitRequestToBackgroundAndCatch(
+  method: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  args?: any[],
+) {
+  submitRequestToBackground(method, args).catch((error) => {
```

Comment:
Looks like we might need optional chaining here?
> https://github.com/MetaMask/metamask-extension/actions/runs/11369849727/job/31628250652?pr=27735

```suggestion
  submitRequestToBackground(method, args)?.catch((error) => {
```
Which is weird...

Reply:
Ah thanks, fixed!  Is it because `submitRequestToBackground` returns a `Promise<R>`?  What's an `R` anyway?

Reply:
That's the type that the promise returned by `submitRequestToBackground` would resolve to. 

(`R` isn't the best name. We have eslint rules for more descriptive generic parameter names, but again it's not being enforced in extension for now.)

However, the TypeScript types shouldn't have any effect here. Despite the misleading `TypeError:` label, that error is purely based on the runtime javascript code.